.github/workflows/validate-pr.yaml

```diff
@@ -129,3 +129,61 @@ jobs:
       - name: Run npm format
         run: |
           cd web && npm run format
+  run-k8s-deployment:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
+      - name: Start minikube
+        uses: meroxa/setup-minikube@master
+        with:
+          minikube-version: 1.24.0
+      - name: run k8s deployment on the cluster
+        run: |
+          namespace=monocle
+          kubectl create namespace "${namespace}" 
+          kubectl create --namespace "${namespace}" --filename ./deployment/ 
+
+          deploy_pod_labels_raw=$(kubectl get --namespace "${namespace}" --filename ./deployment/elastic-deployment.yaml --output go-template='{{ range $k, $v :=.spec.selector.matchLabels }}{{$k}}={{$v}},{{ end }}')
+          deploy_pod_labels_elastic=$(echo "${deploy_pod_labels_raw}" | rev | cut --delimiter=',' --fields='2-' | rev)
+
+          while [[ $(kubectl get po --namespace "${namespace}" --no-headers 2>/dev/null | wc -l) -eq  0 ]]; do
+            echo -n "."
+            sleep 1s
+          done
+          echo
+
+          kubectl wait --for=condition=Ready --namespace "${namespace}" pods --selector "${deploy_pod_labels_elastic}" --timeout=1m
+          kubectl wait --for=condition=Ready --namespace "${namespace}" pods --all
+
+          kubectl create --namespace "${namespace}" job test  --from $( kubectl get --namespace ${namespace} -f ./deployment/crawler-legacy-cronjob.yaml --output name) 
+          kubectl wait --for=condition=complete --namespace "${namespace}" job/test 
+
+      - name: Display indexes
+        run: |
+          namespace=monocle
+          kubectl port-forward --namespace "${namespace}" $( kubectl get --namespace "${namespace}" -f deployment/elastic-service.yaml --output name) 9200:9200 &
+          sleep 5s
+
+          curl -s -I -X GET http://localhost:9200/_cat/indices
+
+          kill %1
+
+#     # this is commented as on k8s there is no nginx to route traffic
```

Comment:
The web container is a nginx server that serves the web app (js) and the API via a proxy_pass option.
So you should be able to test it the same way it is done in .github/valide-pr.yaml:
```
      - name: Check api service through nginx
        run: "curl -s --fail -H 'Content-type: application/json' http://localhost:8080/api/2/get_workspaces -d '{}' | grep 'workspaces'"
```

Reply:
changed and now it works. resolving